### DISCUSSION TITLE
Next JS dynamic routes not working with firebase in production

### DISCUSSION THREAD

**Discussion by talkcurrent** [vercel/next.js#52120] [App Router]
next JS breaking long path in the new 13.4.7 version with firebase in production only

**Comment by icyJoseph** [comment]
Yes, but we'd need to see some example of your file tree, and path.

**Reply by talkcurrent** [reply]
![Screenshot 2023-07-03 at 11 37 58 AM](https://github.com/vercel/next.js/assets/115302680/245808b2-cdfb-4d7b-9fd4-c56eddb6ae89)
![Screenshot 2023-07-03 at 11 38 49 AM](https://github.com/vercel/next.js/assets/115302680/28ddd3a2-b0f3-412d-83c9-01fe4dc467c0)

**Comment by talkcurrent** [comment]
> Yes, but we'd need to see some example of your file tree, and path.

In development mode "npm run dev", everything is working fine but not in production

**Reply by icyJoseph** [reply]
Mmm what's not working? Do you get 404 or something?

**Reply by talkcurrent** [reply]
Yes I get 404 | page not found. 
Expected https://butreport.web.app/reports/fmcb/090666666666 to load the page.jsx 
where: 090666666666 is a dynamic folder as shown in the image 
![Screenshot 2023-07-03 at 2 45 14 PM](https://github.com/vercel/next.js/assets/115302680/57221386-5fda-4b53-97a4-4c419655f91d)

but am getting 404 with a shortened href  "https://butreport.web.app/reports/fmcb/090666" instead of "https://butreport.web.app/reports/fmcb/090666666666 "

Note that this error does not occur in testing mode but after deploying to firebase

**Reply by icyJoseph** [reply]
Oh wow, well, do you have `getStaticParams` and `dynamicParams` configurations for this page? 

Just one more question, are you also using the `export` output mode?

**Reply by talkcurrent** [reply]
![Screenshot 2023-07-03 at 4 57 44 PM](https://github.com/vercel/next.js/assets/115302680/cf702d82-3404-43f1-86a0-5920c09cf77d)
![Screenshot 2023-07-03 at 4 58 04 PM](https://github.com/vercel/next.js/assets/115302680/bed50980-8c74-485e-b222-34797dbf7952)
![Screenshot 2023-07-03 at 4 59 57 PM](https://github.com/vercel/next.js/assets/115302680/02f9bd1a-dad4-406a-816b-36d3717256f9)

Am using export output mode as shown in the image above.
But getStaticParams and dynamicParams configurations.... i haven't done these.

**Reply by icyJoseph** [reply]
Mmm https://github.com/vercel/next.js/issues/51845 does this look like your issue? 

I'm also not sure that the export mode allows dynamic runtime params just yet.

**Reply by talkcurrent** [reply]
Am thinking it's a firebase specific problem.
NextJS Dynamic Routing with Firebase #4980
